src/android/FileTransferBackground.java

```diff
@@ -294,16 +336,30 @@ public void deletePendingUploadAndSendEvent(JSONObject obj) {
         createAndSendEvent(obj);
     }
 
+    private UploadNotificationStatusConfig buildNotification(String title){
+        Activity mainActivity = cordova.getActivity();
+        Resources activityRes = mainActivity.getResources();
+        int iconId = activityRes.getIdentifier("ic_upload", "drawable", mainActivity.getPackageName());
+        Intent intent = new Intent(cordova.getContext(), mainActivity.getClass());
+        PendingIntent clickIntent = PendingIntent.getActivity(cordova.getContext(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+        return new UploadNotificationStatusConfig(
+                title,
+                "",
+                iconId,
+                Color.parseColor("#396496"),
+                null,
+                clickIntent,
+                new ArrayList<>(0),
+                true,
+                true
+        );
+    }
     private UploadNotificationConfig getNotificationConfiguration(String title) {
-        UploadNotificationConfig config = new UploadNotificationConfig();
-        config.getCompleted().autoClear = true;
-        config.getCancelled().autoClear = true;
-        config.getError().autoClear = true;
-        config.setClearOnActionForAllStatuses(true);
-        Intent intent = new Intent(cordova.getContext(), cordova.getActivity().getClass());
-        PendingIntent pendingIntent = PendingIntent.getActivity(cordova.getContext(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
-        config.setClickIntentForAllStatuses(pendingIntent);
-        config.getProgress().title = title;
+        UploadNotificationStatusConfig progress = buildNotification(title);
+        UploadNotificationStatusConfig success = buildNotification("");
+        UploadNotificationStatusConfig error = buildNotification("");
+        UploadNotificationStatusConfig cancelled = buildNotification("");
+        UploadNotificationConfig config = new UploadNotificationConfig("com.spoon.backgroundfileupload.channel", false, progress, success, error, cancelled);
```

Comment:
@mevinDhun Why not ? (ton ide te dit quel argument est quoi non ?

```suggestion
        UploadNotificationConfig config = new UploadNotificationConfig(
          "com.spoon.backgroundfileupload.channel",
           false,
           buildNotificationStatusConfig(title),
           buildNotificationStatusConfig(),
           buildNotificationStatusConfig(),
           buildNotificationStatusConfig()
         );
```